Userland/sleep.cpp

```diff
@@ -24,16 +24,48 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <AK/String.h>
+#include <AK/Optional.h>
 #include <signal.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
+#include <time.h>
 #include <unistd.h>
 
+// The maximum integer we can definitely exactly represent as a double (2^52).
+// Note that clock_nanosleep can handle even larger numbers (up to around 2^63),
+// but I think it's fine to cap silently at a hundred million years.
+static const double MAX_SLEEPABLE = 4503599627370496;
+
 void handle_sigint(int)
 {
 }
 
+double parse_amount(char* arg, bool& ok) {
+    // This can't go into AK/* because AK can't run anything from stdlib,
+    // so strtod is not available yet.
+    // TODO: Find a way to put this function into AK/
```

Comment:
How about we move the bulk of `strtod()` into `AK` and make LibC's `strtod()` a thin wrapper around it? :)

Reply:
Sure! I thought that would go beyond the scope of a single PR.

Reply:
Yeah you can do that in a separate PR ofc :)

Reply:
I would like to continue this issue. Code porting to master is quite trivial. Problem is the arg parsing, which is being worked on in https://github.com/SerenityOS/serenity/pull/2867. 

Technically - how to proceed ? Should I create a new PR for this issue, or my PR should be against this PR, and @BenWiederhake  should handle it? (I think closing this one and me handling it giving all credit to @BenWiederhake  is easier)